flutter/lib/src/integrations/native_app_start_integration.dart

```diff
@@ -149,6 +149,25 @@ class NativeAppStartIntegration extends Integration<SentryFlutterOptions> {
           nativeSpanTimes: nativeSpanTimes);
 
       setAppStartInfo(appStartInfo);
+
+      // When we don't have a SentryNavigatorObserver, a TTID transaction
+      // is not created therefore we need to create a transaction ourselves.
+      // We detect this by checking if the currentRouteName is null.
+      // This is a workaround since there is no api that tells us if
+      // the navigator observer exists or not. The currentRouteName is always
+      // set during a didPush triggered by the navigator observer.
+      if (SentryNavigatorObserver.currentRouteName == null) {
```

Comment:
we could even have an additional static flag `isAttached` that is set to true when the class is instantiated, and use it together with this check, but not sure if that would actually help some case.
Your call here

Reply:
I think naming it something like `isCreated` is more accurate since you can also instantiate it earlier and attach it much later